business/workloads.go

```diff
@@ -1437,14 +1437,16 @@ func (in *WorkloadService) fetchWorkloadsFromCluster(ctx context.Context, cluste
 
 		if cnFound {
 			// Add the Proxy Status to the workload
+			isWaypoint := w.IsWaypoint()
 			for _, pod := range w.Pods {
-				isWaypoint := w.IsWaypoint()
 				if in.conf.ExternalServices.Istio.IstioAPIEnabled && (pod.HasIstioSidecar() || isWaypoint) {
 					pod.ProxyStatus = in.businessLayer.ProxyStatus.GetPodProxyStatus(cluster, namespace, pod.Name, !isWaypoint)
 				}
-				// Add the Proxy Status to the workload
+				// Add the Waypoint Workloads to the workload
```

Comment:
Maybe `AmbientEnabled` should also be a workload level field? I'm guessing it just pulls the redirect annotation which is on the pods and not the workloads themselves.